ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerview.js

```diff
@@ -60,36 +60,53 @@
         }
 
         updateValue(value) {
-            if (this.widgetObject) {
+            if (value && value.trim() != '' && this.widgetObject) {
                 if (this.isActive()) {
                     this.widgetObject.setValue(value);
                 } else {
+                    this._model.value = this.widgetObject.getValue();
                     this.widgetObject.setDisplayValue(value);
                 }
             } else {
                 super.updateValue(value);
             }
         }
 
+        getFormattedDate(value) {
+            if(!value || value.trim() === '') {
+                return '';
+            }
+            let editFormat = this._model.editFormat;
+            if(editFormat === 'date|short') {
+                editFormat = 'date|yyyy/mm/dd';
+            }
+            let currDate = FormView.Formatters.parse(value.toString(), this._model.locale, editFormat, null, false);
+            if (currDate && !isNaN(currDate) && value != null) {
+                return currDate.getFullYear() + "-" + (currDate.getMonth() + 1) +"-"+ currDate.getDate() + "";
+            } else return value;
+        }
+
         setModel(model) {
             super.setModel(model);
             if (!this.#noFormats()) {
                 if (this.widgetObject == null) {
                     this.widgetObject = new DatePickerWidget(this, this.getWidget(), model);
                 }
                 if (this.widgetObject.getValue() !== '') {
-                    this._model.value = this.widgetObject.getValue();
+                    this._model.value = this.getFormattedDate(this.widgetObject.getValue());
```

Comment:
Here, you are basically trying to get the data value which is in ISO format from a formatted value. We should have a function for this in af-core and direcly use it, rather than copying the entire logic of formatters here ? @vdua IIRC, we do have some utility for this  ?

Reply:
Not yet, but I asked @egmox to create it in runtime. I would suggest making editValue writable property and set this._model.editValue =  user entered value
and the setter should parse it accordingly.